src/gluonts/ev/evaluator.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import (
+    Callable,
+    Dict,
+    Optional,
+    Protocol,
```

Comment:
`Protocol` was added in Python 3.8.

Thus we need to import it from `typing_extensions`, as well as `runtime_checkable`.